infrastructure/projet_doctoral/doctorat/epreuve_confirmation/domain/service/notification.py

```diff
@@ -63,20 +73,98 @@ def _get_doctorate_title_translation(cls, doctorate: DoctorateProxy):
         return lazy(lambda: doctorate_title[get_language()], str)()
 
     @classmethod
-    def get_common_tokens(cls, doctorate: DoctorateProxy, confirmation_paper: EpreuveConfirmation):
+    def get_admission_link_back(cls, uuid, tab='project'):
+        return (
+            "{}{}".format(
+                settings.ADMISSION_BACKEND_LINK_PREFIX.rstrip('/'),
+                resolve_url('admission:doctorate:{}'.format(tab), uuid=uuid),
+            ),
+        )
+
+    @classmethod
+    def get_admission_link_front(cls, uuid, tab=''):
+        return settings.ADMISSION_FRONTEND_LINK.format(uuid=uuid) + tab
+
+    @classmethod
+    def get_common_tokens(cls, doctorate, confirmation_paper: Union[EpreuveConfirmationDTO, EpreuveConfirmation]):
         """Return common tokens about a doctorate"""
+        financing_type = (
+            BourseRecherche.get_value(doctorate.scholarship_grant)
+            if doctorate.financing_type == ChoixTypeFinancement.SEARCH_SCHOLARSHIP.name
+            else ChoixTypeFinancement.get_value(doctorate.financing_type)
+        )
+
         return {
             "student_first_name": doctorate.candidate.first_name,
             "student_last_name": doctorate.candidate.last_name,
             "doctorate_title": cls._get_doctorate_title_translation(doctorate),
-            "admission_link_front": settings.ADMISSION_FRONTEND_LINK.format(uuid=doctorate.uuid),
-            "admission_link_back": "{}{}".format(
-                settings.ADMISSION_BACKEND_LINK_PREFIX.rstrip('/'),
-                resolve_url('admission:doctorate:project', pk=doctorate.uuid),
-            ),
+            "admission_link_front": cls.get_admission_link_front(doctorate.uuid),
+            "admission_link_back": cls.get_admission_link_back(doctorate.uuid),
+            "confirmation_paper_link_front": cls.get_admission_link_front(doctorate.uuid, 'confirmation'),
+            "confirmation_paper_link_back": cls.get_admission_link_back(doctorate.uuid, 'confirmation'),
             "confirmation_paper_date": cls.format_date(confirmation_paper.date),
+            "confirmation_paper_deadline": cls.format_date(confirmation_paper.date_limite),
+            "scholarship_grant_acronym": financing_type,
+            "reference": doctorate.reference,
+            "extension_request_proposed_date": confirmation_paper.demande_prolongation.nouvelle_echeance
+            if confirmation_paper.demande_prolongation
+            else '',
         }
 
+    @classmethod
+    def _send_notification_to_managers(cls, entity_id, content, tokens):
+        cdd_managers = CddManager.objects.filter(
+            entity_id=entity_id,
+        ).select_related('person')
+
+        for manager in cdd_managers:
+            final_content = content % tokens
+            with translation.override(manager.person.language):
+                web_notification = WebNotification(recipient=manager.person, content=str(final_content))
```

Comment:
```suggestion
            with translation.override(manager.person.language):
                final_content = content % tokens
                web_notification = WebNotification(recipient=manager.person, content=str(final_content))
```

J'ai déjà eu le problème : le lazy est résolu au `%`